dot/parachain/candidate-validation/candidate_validation.go

```diff
@@ -72,46 +82,37 @@ func (cv *CandidateValidation) Stop() {
 }
 
 // processMessages processes messages sent to the CandidateValidation subsystem
-func (cv *CandidateValidation) processMessages(wg *sync.WaitGroup) {
-	defer wg.Done()
-	for {
-		select {
-		case msg := <-cv.OverseerToSubsystem:
-			logger.Debugf("received message %v", msg)
-			switch msg := msg.(type) {
-			case ValidateFromChainState:
-				// TODO: implement functionality to handle ValidateFromChainState, see issue #3919
-			case ValidateFromExhaustive:
-				result, err := validateFromExhaustive(cv.ValidationHost, msg.PersistedValidationData,
-					msg.ValidationCode, msg.CandidateReceipt, msg.PoV)
-				if err != nil {
-					logger.Errorf("failed to validate from exhaustive: %w", err)
-					msg.Ch <- parachaintypes.OverseerFuncRes[ValidationResult]{
-						Err: err,
-					}
-				} else {
-					msg.Ch <- parachaintypes.OverseerFuncRes[ValidationResult]{
-						Data: *result,
-					}
-				}
-
-			case PreCheck:
-				// TODO: implement functionality to handle PreCheck, see issue #3921
-
-			case parachaintypes.ActiveLeavesUpdateSignal:
-				_ = cv.ProcessActiveLeavesUpdateSignal(msg)
-
-			case parachaintypes.BlockFinalizedSignal:
-				_ = cv.ProcessBlockFinalizedSignal(msg)
-
-			default:
-				logger.Errorf("%w: %T", parachaintypes.ErrUnknownOverseerMessage, msg)
+func (cv *CandidateValidation) processMessages(msg any) error {
+	switch msg := msg.(type) {
+	case ValidateFromChainState:
+		// TODO: implement functionality to handle ValidateFromChainState, see issue #3919
+	case ValidateFromExhaustive:
+		result, err := validateFromExhaustive(cv.ValidationHost, msg.PersistedValidationData,
+			msg.ValidationCode, msg.CandidateReceipt, msg.PoV)
+		if err != nil {
+			logger.Errorf("failed to validate from exhaustive: %w", err)
+			msg.Ch <- parachaintypes.OverseerFuncRes[ValidationResult]{
+				Err: err,
+			}
+		} else {
+			msg.Ch <- parachaintypes.OverseerFuncRes[ValidationResult]{
+				Data: *result,
 			}
-
-		case <-cv.stopChan:
-			return
 		}
+
+	case PreCheck:
+		// TODO: implement functionality to handle PreCheck, see issue #3921
```

Comment:
Maybe would be nice to add panic with this TODO message?

Reply:
updated